packages/mock-block-dock/dev/dev-app.tsx

```diff
@@ -1,42 +1,61 @@
-import { Entity } from "@blockprotocol/graph";
+import { extractBaseUri } from "@blockprotocol/type-system/slim";
 import { ChangeEvent, FunctionComponent, useState } from "react";
 import { createRoot } from "react-dom/client";
 
 import { MockBlockDock } from "../src";
+import { entityTypes } from "../src/data/entity-types";
+import { propertyTypes } from "../src/data/property-types";
 import { TestCustomElementBlock } from "./test-custom-element-block";
 import testBlockString from "./test-html-block/block.html?raw";
 import { TestReactBlock } from "./test-react-block";
 
 const node = document.getElementById("app");
 
-type TestBlockType =
-  | "react"
-  | "custom-element"
-  | "html-at-url"
-  | "html-as-string";
-
 const blockEntityMap = {
   react: {
-    entityId: "entity-react",
-    entityTypeId: "test-type-1",
-    properties: { name: "World" },
+    metadata: {
+      editionId: {
+        baseId: "entity-react",
+        versionId: new Date().toISOString(),
```

Comment:
Feels a bit odd to have `versionId` part of this when it is unused .3

Reply:
`baseId` is `EntityId`

So you have 
```ts
type EntityEditionId: {
  baseId: EntityId,
  recordId: string
}
```

Reply:
> and the difference between baseId and recordId
This: 

> No, because blocks will need the knowledge about what identifies an entity, and what identifies a specific edition of an entity. 

The `EntityId` establishes the _identity_ of an entity (e.g. "Bob"), it doesn't identify a specific edition of an entity.

We need two pieces of information, what is the thing, and what edition of that thing is it.

We previously were using version to identify that edition, but if we want to support multi-temporal-axis systems then a singular version string will not be good enough to identify the specific record we're talking about.

So we need 
- `EntityId` the identity of the thing we're talking about
- `X` the identity of the exact record that we're looking at here, a specific edition of the thing
- `firstAxisTime`, the version of this on the first axis 
- `secondAxisTime`, the version of this on the second axis

Where `firstAxisTime` and `secondAxisTime` are not part of the record's identifier, so they can live inside `entity.metadata.version` or something.

Reply:
X _opaquely_ defines a specific `(firstAxisTime, secondAxisTime)` combination of a record. You can think of it as the primary key in the data base.

Reply:
I would keep X as `versionId` then, because 
(a) it makes it more obvious that it is irrelevant to non-versioned systems (is it omitted in these cases? any old junk passed?)
(b) in non-versioned systems, the id of the record is possibly identical to the `baseId` (I'd also consider renaming this – why not `entityId`?)

Reply:
If we are saying that X is _only required to distinguish a specific version of a thing_

in non-versioned systems this description:
> just an opaque unique identifier for the specific record in the DB

will be `baseId` (as it is currently called)

Reply:
Let's jump on a call tomorrow or something about this then, I think I'm not conveying this well, we physically cannot combine them, and `versionId` is misleading because versions are distinguished by their timestamps, and the whole multi-axis thing makes that confusing. Hence recordId, the whole point is to not misleadingly convey meaning about temporal versioning within it, it's just a necessary piece of metadata in some systems, and in other systems they can just set it to any string they like